elixir/native/extism_nif/src/lib.rs

```diff
@@ -0,0 +1,89 @@
+use rustler::{Error, Atom};
+use extism::{Plugin};
+use std::str;
+use serde_json;
+use std::path::Path;
+use std::str::FromStr;
+
+mod atoms {
+    rustler::atoms! {
+        ok,
+        error,
+        eof,
+        unknown // Other error
+    }
+}
+
+// fn load(env: Env, _: Term) -> bool {
+//     true
+// }
+
+ #[rustler::nif]
+fn plugin_new_with_manifest(manifest_payload: String, wasi: bool) -> Result<isize, Error> {
+    match serde_json::from_str(&manifest_payload) {
```

Comment:
Instead of parsing should I just pass the rendered json right down to the SDK?

Reply:
Yea I'd go straight to the SDK. The `Result` still can be handled nicely in Elixir I think, which is returned from the `Plugin::new_with_manifest` anyways.